src/main/java/bio/terra/cli/apps/CommandRunner.java

```diff
@@ -0,0 +1,136 @@
+package bio.terra.cli.apps;
+
+import static bio.terra.cli.service.WorkspaceManager.getAiNotebookInstanceName;
+import static bio.terra.cli.service.WorkspaceManager.getBigQueryDatasetPath;
+import static bio.terra.cli.service.WorkspaceManager.getGcsBucketUrl;
+
+import bio.terra.cli.command.exception.SystemException;
+import bio.terra.cli.context.GlobalContext;
+import bio.terra.cli.context.WorkspaceContext;
+import bio.terra.workspace.model.ResourceDescription;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public abstract class CommandRunner {
```

Comment:
There's already a comment for this class. I pushed a commit with some missing comments ~20 min after I requested a review (I'd forgotten to push), so maybe you didn't see it. Let me know if there's something else to add to the comment.

Reply:
I missed the commit, but I think it might still be good to add a little more about why this is useful regarding setting config/env variables in this base class vs what the subclasses do.

Reply:
added more details.